napari/layers/shapes/_shapes_utils.py

```diff
@@ -736,22 +736,27 @@ def generate_2D_edge_meshes(path, closed=False, limit=3, bevel=False):
     )[0]
 
     if len(idx_bevel) > 0:
-        # only the 'outwards sticking' offsets should be changed
-        # TODO: This is not entirely true as in extreme cases both can go to infinity
         idx_offset = (miter_signs[idx_bevel] < 0).astype(int)
-        idx_bevel_full = 2 * idx_bevel + idx_offset
+
+        # outside and inside offsets are treated differently (only outside offsets get beveled)
+        idx_bevel_outside = 2 * idx_bevel + idx_offset
+        idx_bevel_inside = 2 * idx_bevel + (1 - idx_offset)
         sign_bevel = np.expand_dims(miter_signs[idx_bevel], -1)
 
-        # adjust offset of outer "left" vertex
-        offsets[idx_bevel_full] = (
+        # adjust offset of outer offset
+        offsets[idx_bevel_outside] = (
             -0.5 * full_normals[:-1][idx_bevel] * sign_bevel
         )
+        # adjust offset of inner offset
+        offsets[idx_bevel_inside] = (
+            0.5 * full_normals[:-1][idx_bevel] * sign_bevel
+        )
```

Comment:
@maweigert this looks good to me in that it doesn't look alarming 😂, but I don't actually understand it. I'm wondering if you could draw a doodle about what's going on in this code, and take a photo and add it to a comment in this PR, and then we can link to that comment in this code so that future generations can understand it. 😅

Reply:
<img width="873" alt="image" src="https://github.com/napari/napari/assets/11042162/683c185f-c3a9-4a52-bcaf-61d0c5caf7b7">

Reply:
Thanks! I had gathered that much before, but "adjust offset of inner offset" is confusing, since it doesn't look like there's anything to adjust? Or does "adjust" mean "scale" here? Or...?

Reply:
You are right, this line 747 was a bit funky. Indeed, when I tried to visualize it now I noticed that the "fix" was actually wrong! Whereas the outer offset (the one at the new bevel triangle) needs to set to the normal to make room for the new vertex, the inner offset should only be scaled such to avoid any edge explosion. Thanks a lot for insisting on me explaining :) I updated the PR...

Reply:
Heh, thanks, but now it looks like the test data needs to be updated again. 😅🙏